shared/src/commonTest/kotlin/shared/internal/database/DatabaseTests.kt

```diff
@@ -0,0 +1,85 @@
+package shared.internal.database
+
+import app.cash.turbine.test
+import com.automattic.myapplication.shared.TracksEvent
+import com.automattic.myapplication.shared.internal.SqlDelightRepository
+import com.automattic.myapplication.shared.internal.database.Database
+import com.automattic.myapplication.shared.internal.database.DriverFactory
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.runBlocking
+import kotlin.test.assertTrue
+
+internal class DatabaseTests {
```

Comment:
The whole idea for `DatabaseTests` is to test SQLDelight on each platform (Android and iOS) with its native driver.

We have to create a native driver in platform-specific source sets as it requires platform specific dependencies, like `Context` in Android. So we create it there and then run `DatabaseTests`.

The cons of this approach are less readable test results as we technically have only one `@Test` (that will run multiple tests) but we can see which test failed in the stacktrace which is not the worse thing ever.

The other idea would be to create a new, [test-only multiplatform module like KaMPKit does](https://github.com/touchlab/KaMPKit/blob/f0e90ccb71deead9bd6cb496ad960bd5d9c7c83c/shared/src/commonTest/kotlin/co/touchlab/kampkit/SqlDelightTest.kt), but it'd require us to remove `internal` from classes we want to be `internal` and this would cause a lot of problems for consumers I believe.